test/Microsoft.ML.Tests/CopyColumnEstimatorTests.cs

```diff
@@ -0,0 +1,128 @@
+﻿using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.ML.Tests
+{
+    public class CopyColumnEstimatorTests
+    {
+        class TestClass
+        {
+            public int A;
+            public int B;
+            public int C;
+        }
+        class TestClassXY
+        {
+            public int X;
+            public int Y;
+        }
+
+
+        [Fact]
+        void TestWorking()
+        {
+            var data = new List<TestClass>() { new TestClass() { A = 1, B = 2, C = 3, }, new TestClass() { A = 4, B = 5, C = 6 } };
+            using (var env = new TlcEnvironment())
+            {
+                var dataView = ComponentCreation.CreateDataView(env, data);
+                var est = new CopyColumnsEstimator(env, new[] { ("A", "D"), ("B", "E") });
+                var transformer = est.Fit(dataView);
+                var result = transformer.Transform(dataView);
+                ValidateCopyColumnTransformer(result);
+            }
+        }
+
+        private void ValidateCopyColumnTransformer(IDataView result)
+        {
+            using (var cursor = result.GetRowCursor(x => true))
+            {
+                DvInt4 avalue = 0;
+                DvInt4 bvalue = 0;
+                DvInt4 dvalue = 0;
+                DvInt4 evalue = 0;
+                var aGetter = cursor.GetGetter<DvInt4>(0);
+                var bGetter = cursor.GetGetter<DvInt4>(1);
+                var dGetter = cursor.GetGetter<DvInt4>(3);
+                var eGetter = cursor.GetGetter<DvInt4>(4);
+                while (cursor.MoveNext())
+                {
+                    aGetter(ref avalue);
+                    bGetter(ref bvalue);
+                    dGetter(ref dvalue);
+                    eGetter(ref evalue);
+                    Assert.Equal(avalue, dvalue);
+                    Assert.Equal(bvalue, evalue);
+                }
+            }
+        }
+
+        [Fact]
+        void TestBadOriginalSchema()
+        {
+            var data = new List<TestClass>() { new TestClass() { A = 1, B = 2, C = 3, }, new TestClass() { A = 4, B = 5, C = 6 } };
+            using (var env = new TlcEnvironment())
+            {
+                var dataView = ComponentCreation.CreateDataView(env, data);
+                var est = new CopyColumnsEstimator(env, new[] { ("D", "A"), ("B", "E") });
+                bool failed = false;
+                try
+                {
+                    var transformer = est.Fit(dataView);
+                }
+                catch
+                {
+                    failed = true;
+                }
+                Assert.True(failed);
```

Comment:
>Assert [](start = 16, length = 6)

can you just make it `Assert.True(false)` inside the try block?

Reply:
I reject your rejection. Just put Assert.False() inside the try (not catch) block after the code that is supposed to throw

---
In reply to: [212474277](https://github.com/dotnet/machinelearning/pull/706#discussion_r212474277) [](ancestors = 212474277,211795480)